libc-bottom-half/sources/__wasilibc_real.c

```diff
@@ -365,228 +340,214 @@ __wasi_errno_t __wasi_fd_tell(
     __wasi_fd_t fd,
     __wasi_filesize_t *retptr0
 ){
-    int32_t ret = __imported_wasi_snapshot_preview1_fd_tell((int32_t) fd, (int32_t) retptr0);
-    return (uint16_t) ret;
+    return (__wasi_errno_t) __imported_wasi_snapshot_preview1_fd_tell((uint32_t) fd, (uintptr_t) retptr0);
 }
 
-int32_t __imported_wasi_snapshot_preview1_fd_write(int32_t arg0, int32_t arg1, int32_t arg2, int32_t arg3) __attribute__((
+uint32_t __imported_wasi_snapshot_preview1_fd_write(uint32_t, uintptr_t, __wasi_size_t, uintptr_t) __WASI_NOEXCEPT __attribute__((
     __import_module__("wasi_snapshot_preview1"),
     __import_name__("fd_write")
 ));
 
 __wasi_errno_t __wasi_fd_write(
     __wasi_fd_t fd,
-    const __wasi_ciovec_t *iovs,
+    __wasi_ciovec_t const *iovs,
     size_t iovs_len,
     __wasi_size_t *retptr0
 ){
-    int32_t ret = __imported_wasi_snapshot_preview1_fd_write((int32_t) fd, (int32_t) iovs, (int32_t) iovs_len, (int32_t) retptr0);
-    return (uint16_t) ret;
+    return (__wasi_errno_t) __imported_wasi_snapshot_preview1_fd_write((uint32_t) fd, (uintptr_t) iovs, (__wasi_size_t) iovs_len, (uintptr_t) retptr0);
 }
 
-int32_t __imported_wasi_snapshot_preview1_path_create_directory(int32_t arg0, int32_t arg1, int32_t arg2) __attribute__((
+uint32_t __imported_wasi_snapshot_preview1_path_create_directory(uint32_t, uintptr_t, __wasi_size_t) __WASI_NOEXCEPT __attribute__((
```

Comment:
Why are all tese `int32_t` -> `uint32_t` changes needed?

Reply:
signed integer can easily trigger undefined behavior while unsigned integer will only wrap but not undefined behavior.

Reply:
ok